Scenario Mode

Tokiko1 opened the pull request:
Birds and maids.

Tokiko1 committed:
Adds some basic config file stuff for scenario mode.

Tokiko1 committed:
Custom jobs and role added restriction to config.

Tokiko1 committed:
Morework.

imtakingabreakdontatme commented on code/game/gamemodes/scenario/scenario.dm:
```diff
@@ -0,0 +1,44 @@
+
+/datum/game_mode/scenario/proc/pickscenario()
+	var/list/datum/scenario/scenario_list = subtypesof(/datum/scenario)
+	choosen_scenario = pick(scenario_list)
```
var/scenario_type = pick(scenario_list)

chosen_scenario = new scenario_type

Tokiko1 committed:
Fixes. Thank you Kor!

Tokiko1 committed:
Adds faction assigning and faction role restriction functionality.

Tokiko1 committed:
Adds fluff text displaying functionality.

Tokiko1 committed:
Adds role/faction item spawning for players.

Tokiko1 committed:
Not related to scenario mode but who is going to stop me?
Improves the white dress further and adds a fancy black dress and gloves.
Uniforms now render above shoes.

imtakingabreakdontatme commented:
>Not related to scenario mode but who is going to stop me? 

Please atomize your PRs or I'll be forced to reject it

Tokiko1 committed:
Adds goal checking functionality for mob based objectives.
Part 1 of 5(?) of goal checking system.

Tokiko1 committed:
Part 2/5(?) of the goal autochecking system...
Also added goalchecking proc into the config, perhaps that is more sane than my messy, lazy idea?

Also makes harpy headwings look better.

Tokiko1 committed:
Adds subscenarios to make scenarios easier to RP out.
Adds the banquet subscenario.

Tokiko1 committed:
Adds the guest job and makes goal checking less insane and adds round-start goal announcement.

Tokiko1 committed:
Adds limits for factions.

Tokiko1 committed:
Adds custom vote functionality.

Tokiko1 committed:
Adds gamemode processing and endround handling stuff. 1/2

Tokiko1 committed:
Adds roundend input handling and poll handling 2/2 and it works.

Tokiko1 committed:
ALL DONE!
